manticore/ethereum/__init__.py

```diff
@@ -99,29 +100,28 @@ def will_open_transaction_callback(self, state, tx):
             #Let's compile  the list of interesting hashes
             selected_functions = []
 
-            for func_hsh in md.hashes:
-                if func_hsh == b'\0\0\0\0':
-                    continue
+            for func_hsh in md.function_selectors:
                 abi = md.get_abi(func_hsh)
-                func_name = md.get_func_name(func_hsh)
+                if abi['type'] == 'fallback':
+                    continue
                 if self._mutability == 'constant' and not abi.get('constant', False):
                     continue
                 if self._mutability == 'mutable' and abi.get('constant', False):
                     continue
-                if not re.match(self._regexp, func_name):
+                if not re.match(self._regexp, abi['name']):
```

Comment:
Loud thinking: is using regex here a good idea? It might slow down things a lot (since this is a callback).

@feliam @yan @mossberg what do u think?